storm-core/src/jvm/org/apache/storm/state/KeyValueState.java

```diff
@@ -45,4 +47,9 @@
      * @return the value or defaultValue if no mapping is found
      */
     V get(K key, V defaultValue);
+
+    /**
+     * @return all key value mappings as an unmodifiable map.
+     */
+    Map<K, V> getAll();
```

Comment:
cool. I wasn't aware `hscan` gives you a `cursor`, redis is sort of new to me. I am gonna look into it and will update the pr. thanks guys. :)